service-base/src/main/java/org/eclipse/hono/service/auth/device/CredentialsApiAuthProvider.java

```diff
@@ -90,7 +90,9 @@ public CredentialsApiAuthProvider(final CredentialsClientFactory credentialsClie
      * @return A future containing the credentials on record as retrieved from
      *         Hono's <em>Credentials</em> API.
      * @throws NullPointerException if device credentials is {@code null}.
+     * @deprecated Use {@link #getCredentialsForDevice(AbstractDeviceCredentials, SpanContext)} instead.
      */
+    @Deprecated
```

Comment:
My gut feeling is that creating a new version of this method is overkill since the method is final as well as the only place in Hono's codebase where it is used (which is in the same class). So the only case in which changing the signature might cause problems is when someone uses it for something completely different. WDYT?

Reply:
My understanding of generics is limited (at best) but IMHO you could simply replace the `DeviceCredentials` parameter type with `T` without breaking any client code because `DeviceCredentials` in fact `extends AbstractDeviceCredentials` ...

Reply:
If I change the type of the parameter to T you could no longer pass in an implementation of the interface `DeviceCredentials` that does not inherit from `AbstractDeviceCredentials`.

Reply:
Ah, I thought `DeviceCredentials` was a class, my bad ...